addons/product/models/product_template.py

```diff
@@ -29,6 +29,9 @@ def _get_default_category_id(self):
 
     @tools.ormcache()
     def _get_default_uom_id(self):
+        uom = self.env.context.get('uom')
+        if uom:
+            return uom
```

Comment:
If the issue is that the default uom used is not the right one, can't you add a `default_uom_id` key in `sale_order_views.xml`, next to the `uom` one ?

I believe we should rely on the standard behavior as much as possible, instead of adding new context keys everywhere.

Reply:
Yes we can but main problem is when we create product on new line where UOM is not set it gets `default_uom_id: False`  in context and since context has more priority then default method on field it set False UOM instead of getting default `Units`  and because of that there some tours failing

So problem with standard behavior is we can't pass context depending on value of UOM on SOL like if SOL has uom set then pass other wise no context
If we really want to use the context and make it work properly then we probably have to override `default_get`  to check result and set uom if uom is false instead `default` method on field